### PR TITLE
Support `swift test`

### PR DESCRIPTION
**Pull request by norio-nomura**
Includes re-organizing `Benchmark` directory.

### CODE REVIEW

**Comment by norio-nomura** [comment]
Now we can run benchmark as:
```sh
% cd Benchmark
% swift build
% swift test  # PerformanceTests with debug configuration
% swift build --configuration release
% .build/release/TryParsecBenchmark
```

**Comment by norio-nomura** [comment]
`Tests/Specs` and `Tests/Helpers` are moved to e `excludedTests` because `Quick` and `Nimble` can't be build by Swift Package Manager.

So, we have no test on running `swift test`! :stuck_out_tongue:

**Comment by chriseidhof** [comment]
Is there a reason why we're not using `measureBlock`? I haven't really used it myself so much, but it could be interesting!

**Comment by norio-nomura** [comment]
Because, I can't find how to run `swift test` with `--configuration release`.

**Comment by inamiy** [comment]
@chriseidhof `measureBlock` test is already used in `JSONPerformanceTests.swift`, and Benchmark is another convenient one for Linux testing.

**Comment by norio-nomura** [comment]
As I tested, building on Linux is failing with Result-1.0.2.
I'm considering to releasing Result-1.0.3 by my self that can be build https://github.com/antitypical/Result/pull/140.

File: Benchmark/Sources/Benchmark/Benchmark.swift [hunk 1]
```diff
@@ -0,0 +1,2 @@
+import Curry
+import TryParsec
```
Comment (inamiy) [inline Benchmark/Sources/Benchmark/Benchmark.swift]: Is there any reason of adding `Curry.framework` for benchmark test?

Also, I'm not sure why this `Benchmark/Benchmark.swift` is needed alongside with `TryParsecBenchmark/main.swift`.

Comment (norio-nomura) [inline Benchmark/Sources/Benchmark/Benchmark.swift]: `TryParsecBenchmark` is executable target.
`Benchmark` is module target that can be imported from `Tests/Benchmark`.
`Tests/Benchmark` is importing `Curry` but we can not specify to link that directory to `Tests/Benchmark`. So, `Benchmark/Benchmark.swift` imports `Curry`

Comment (norio-nomura) [inline Benchmark/Sources/Benchmark/Benchmark.swift]: Oh, I tested it again without `import Curry` and success!
I'll remove that.

Comment (inamiy) [inline Benchmark/Sources/Benchmark/Benchmark.swift]: Yes, `Curry` is actually only used for QuickTests and Playground :wink:

**Comment by norio-nomura** [comment]
Updated.

**Comment by inamiy** [comment]
Oh, I was misunderstanding that performance tests (e.g. `JSONPerformanceTests.swift`) are now moved to `Benchmark` directory. 
In that case, we need `Curry.framework` as (test) dependencies.

I personally think we don't need extra `Benchmark` target and just move `Benchmark/Tests/Benchmark` directory to `Benchmark/Tests/TryParsecBenchmark` as single implicit target.

**Comment by norio-nomura** [comment]
@inamiy You are right!
Updated.

**Comment by inamiy** [comment]
@norio-nomura Thank you for organizing!

**Comment by norio-nomura** [comment]
Thanks! :pray:

**Pull request merged**